src/metadata/persistence/avu.clj

```diff
@@ -1,52 +1,92 @@
   [target-types target-ids attributes values]
-  (select :avus
-          (modifier "DISTINCT")
-          (fields :target_id
-                  :target_type)
-          (where {:target_id   [in target-ids]
-                  :target_type [in (map db/->enum-val target-types)]
-                  :attribute   [in attributes]
-                  :value       [in values]})))
+  (let [cols [:target_id :target_type]
+        q (-> (avus-base-query cols)
+              (dissoc :select) ;; change to select distinct
+              (h/select-distinct cols)
+              (h/where [:in :target_id target-ids]
+                       [:in :target_type (map jtypes/as-other target-types)]
+                       [:in :attribute attributes]
+                       [:in :value values]))]
+    (plan/select! ds cols (sql/format q))))
 
 (defn get-avu-by-id
   "Fetches an AVU by its ID."
   [id]
-  (first (select :avus (where {:id id}))))
+  (let [q (-> (avus-base-query avu-columns)
+              (h/where [:= :id id]))]
+    (plan/select-one! ds avu-columns (sql/format q))))
 
 (defn get-avus-by-ids
   "Finds existing AVUs by a set of IDs."
   [ids]
-  (select :avus (where {:id [in ids]})))
+  (let [q (-> (avus-base-query avu-columns)
+              (h/where [:in :id ids]))]
+    (plan/select! ds avu-columns (sql/format q))))
+
 
 (defn get-avus-for-target
   "Gets AVUs for the given target."
   [target-type target-id]
-  (select :avus (target-where-clause target-type target-id)))
+  (let [q (-> (avus-base-query avu-columns)
+              (h/where [:= :target_id target-id]
+                       [:= :target_type (jtypes/as-other target-type)]))]
+    (plan/select! ds avu-columns (sql/format q))))
 
 (defn get-avus-by-attrs
   "Finds all existing AVUs by the given targets and the given set of attributes."
   [target-types target-ids attributes]
-  (select :avus
-          (where {:attribute   [in attributes]
-                  :target_id   [in target-ids]
-                  :target_type [in (map db/->enum-val target-types)]})))
+  (let [q (-> (avus-base-query avu-columns)
+              (h/where [:in :attribute attributes]
+                       [:in :target_id target-ids]
+                       [:in :target_type (map jtypes/as-other target-types)]))]
+    (plan/select! ds avu-columns (sql/format q))))
+
+(defn add-avus
+  "Adds the given AVUs to the Metadata database."
+  [user-id avus]
+  (let [cols [:id :attribute :value :unit :target_type :target_id :created_by :modified_by]
+        fmt-avu (fn [avu]
+                  (let [avu (-> avu
+                                (update :target_type jtypes/as-other)
+                                (assoc :created_by user-id
+                                       :modified_by user-id))]
+                    (mapv #(get avu %) cols)))]
+    (jsql/insert-multi! ds
+                        (t "avus")
+                        cols
+                        (map fmt-avu avus))))
```

Comment:
The format for these bulk inserts is different than korma. For this, we pass a collection of columns, plus a collection-of-collections, where each entry should be a collection corresponding to the columns listed before.

i.e. `(:id :attribute ...)` and then `[("4609ef48-...." "some-attribute" ...), ...]`